flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -125,7 +142,7 @@ public ExecutionVertex(
 
 		this.inputEdges = new ExecutionEdge[jobVertex.getJobVertex().getInputs().size()][];
 
-		this.priorExecutions = new CopyOnWriteArrayList<Execution>();
+		this.priorExecutions = new EvictingBoundedList<>(maxPriorExecutionHistoryLength);
```

Comment:
Similarly, the ArchivedExecutionVertex is now probably broken if an execution was discarded since the size checks in `getPriorExecutionAttempt` are not adjusted.